### PR TITLE
DLS-5837 FeatureSwitch

### PR DESCRIPTION
**Pull request by JakeReid2020**

### CODE REVIEW

File: app/uk.gov.hmrc.helptosaveproxy/connectors/DWPConnector.scala [hunk 1]
```diff
@@ -68,26 +70,37 @@ class DWPConnectorImpl @Inject()(
 
     val timeContext: Timer.Context = metrics.dwpClaimantCheckTimer.time()
 
+//    val queryParams = Seq(
```
Comment (priyankadhiman3006) [inline app/uk.gov.hmrc.helptosaveproxy/connectors/DWPConnector.scala:73]: commented code

File: conf/application.conf [hunk 2]
```diff
@@ -297,7 +297,7 @@ feature-toggles {
   }
 
   isNewDwpEndpoint{
-    enabled = true
```
Comment (LAJW) [inline conf/application.conf]: What bugs is this commit fixing? Please, update the commit message.

File: app/uk.gov.hmrc.helptosaveproxy/connectors/DWPConnector.scala [hunk 3]
```diff
@@ -119,39 +119,6 @@ class DWPConnectorImpl @Inject()(
...
-//          case e =>
-//            val time = timeContext.stop()
-//            pagerDutyAlerting.alert("Failed to make call to uc claimant check")
-//            metrics.dwpClaimantErrorCounter.inc()
-//            Left(
-//              s"Encountered error while trying to make ucClaimantCheck call, with message: ${e.getMessage}, " +
-//                s"transactionId: $transactionId, thresholdAmount: $threshold, ${timeString(time)}")
-//        })
-//  }
-
```
Comment (LAJW) [inline app/uk.gov.hmrc.helptosaveproxy/connectors/DWPConnector.scala]: :pick: I like to put these into a separate commit (remove commented out tests). Especially since they don't fix any bugs :)

File: app/uk.gov.hmrc.helptosaveproxy/connectors/DWPConnector.scala [hunk 4]
```diff
@@ -68,18 +70,25 @@ class DWPConnectorImpl @Inject()(
...
 
-//    val queryParams = Seq(
-//      "systemId"        -> appConfig.systemId,
-//      "thresholdAmount" -> threshold.toString,
-//      "transactionId"   -> transactionId.toString)
+    val isNewDwpEndpoint = FEATURE("isNewDwpEndpoint", appConfig.runModeConfiguration, logger).enabled
+
+    val dwpUrl = isNewDwpEndpoint match {
+    case true => s"${appConfig.newDwpUrl}/$nino"
+    case _ => s"${appConfig.newDwpUrl}/$nino"
```
Comment (LAJW) [inline app/uk.gov.hmrc.helptosaveproxy/connectors/DWPConnector.scala]: ⚠️ These two lines seem identical. You sure there isn't an error? If not, please combine.
⛏️  Indentataion, format document.

File: app/uk.gov.hmrc.helptosaveproxy/connectors/DWPConnector.scala [hunk 5]
```diff
@@ -106,6 +110,39 @@ class DWPConnectorImpl @Inject()(
         })
   }
 
+//    EitherT(
```
Comment (LAJW) [inline app/uk.gov.hmrc.helptosaveproxy/connectors/DWPConnector.scala]: ⛏️ More commented out code.

**Comment by LAJW** [comment]
This is based on:
https://github.com/hmrc/help-to-save-proxy/tree/dave/DLS-5837/updates

Should we create a PR for it first?

**Comment by LAJW** [comment]
Because of the inclusion of Dave's massive branch (which I can only praise), it is a bit difficult to review (and also hard to revert).

I'd separate this PR into two - one with your changes, and another one with Dave's changes (which I'd approve instantly) and merge them separately, so in case we need to revert something, we only revert one thing and don't cause too much havoc.

Not a fan of commented code. Commit messages aren't descriptive.

There's one bit that's kind of surprising and looks like a typo.

⛏️ - nitpick
